java/src/main/native/include/maps_column_view.hpp

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/column/column.hpp>
+#include <cudf/column/column_view.hpp>
+#include <cudf/lists/lists_column_view.hpp>
+#include <rmm/cuda_stream_view.hpp>
+
+namespace cudf {
+
+class scalar;
+
+namespace jni {
+
+/**
+ * @brief Given a column-view of LIST<STRUCT<K,V>>, an instance of this class
+ * provides an abstraction of a column of maps.
+ *
+ * Each list row is treated as a map of key->value, with possibly repeated keys.
+ * The list may be looked up by a scalar key, or by a column of keys, to
+ * retrieve the corresponding value.
+ */
+class maps_column_view {
+public:
+  maps_column_view(lists_column_view const &lists_of_structs,
+                   rmm::cuda_stream_view stream = rmm::cuda_stream_default);
+
+  // Rule of 5.
+  maps_column_view(maps_column_view const &maps_view) = default;
+  maps_column_view(maps_column_view &&maps_view) = default;
+  maps_column_view &operator=(maps_column_view const &) = default;
+  maps_column_view &operator=(maps_column_view &&) = default;
+  ~maps_column_view() = default;
+
+  /**
+   * @brief Returns number of map rows in the column.
+   */
+  size_type size() const { return keys_.size(); }
+
+  /**
+   * @brief Getter for keys as a list column.
+   *
+   * Note: Keys are not deduped. Repeated keys are returned in order.
+   */
+  lists_column_view keys() const { return keys_; }
```

Comment:
```suggestion
  lists_column_view const& keys() const { return keys_; }
```

Reply:
We should avoid copy as much as possible.

Reply:
Good idea. Fixed.